pyk/src/pyk/kcfg/minimize.py

```diff
@@ -96,26 +95,19 @@ def lift_split_edge(self, b_id: NodeIdLike) -> None:
         # Ensure split can be lifted soundly (i.e., that it does not introduce fresh variables)
         assert (
             len(split_from_b.source_vars.difference(a.free_vars)) == 0
-            and len(split_from_b.target_vars.difference(split_from_b.source_vars)) == 0
+            and len(split_from_b.target_vars.difference(split_from_b.source_vars)) == 0  # <-- Can we delete this check?
         )
         # Create CTerms and CSubsts corresponding to the new targets of the split
-        new_cterms_with_constraints = [
-            (CTerm(a.cterm.config, a.cterm.constraints + csubst.constraints), csubst.constraint) for csubst in csubsts
-        ]
-        # Generate substitutions for new targets, which all exist by construction
-        new_csubsts = [
-            not_none(a.cterm.match_with_constraint(cterm)).add_constraint(constraint)
-            for (cterm, constraint) in new_cterms_with_constraints
-        ]
+        new_cterms_with_constraints = [csubst(a.cterm) for csubst in csubsts]
```

Comment:
Simplified by using `create_split_by_nodes`

Reply:
Let's rename the variable to `new_cterms` then, instead of `new_cterms_with_constraints`.